XenModel/XenAPI-Extensions/Network.cs

```diff
@@ -293,6 +293,11 @@ public bool IsVLAN()
             return Connection.ResolveAll(PIFs).Find(pif => pif.VLAN >= 0) != null;
         }
 
+        public bool IsSriov()
+        {
+            return Connection.ResolveAll(PIFs).Find(pif => pif.sriov_logical_PIF_of != null && pif.sriov_logical_PIF_of.Count != 0) != null;

```

Comment:
As I had also said on PR  #1954, please create a PIF.IsSrIovLogicalPIF() function and use it here as well.